### DISCUSSION TITLE
[@types/react]: Property 'value' does not exist on type 'EventTarget'.

### DISCUSSION THREAD

**Discussion by johann1301s** [DefinitelyTyped/DefinitelyTyped#64682] [Issues with a @types package] [answered]
The type of `onInput` prop on `textarea` element is wrong.

```tsx
const MyComponent = () => {
   return <textarea onInput={(event) => event.target.value}> // Property 'value' does not exist on type 'EventTarget'.ts(2339)
}
```
```json
  "dependencies": {
    "react": "^18.2.0",
    "react-dom": "18.2.0"
  },
  "devDependencies": {
    "@types/react": "18.0.25"
  }
```

The `value` does exist.

[a comment by an automated account is not shown]

**Comment by hitsthings** [comment, marked as answer]
There is a whole lot of discussion at https://github.com/DefinitelyTyped/DefinitelyTyped/pull/12239 and through the rabbit hole from there/

tl;dr for the tl;dr: you should probably use `e.currentTarget` instead of `e.target`

tl;dr:

`e.currentTarget` is the element where the event was attached and you know the type of it.

`e.target` is the element that was actually clicked and it could be any element within `e.currentTarget` - you don't know the type of it EXCEPT if `e.currentTarget` is one of the kinds of nodes that MUST be empty (and then you know that `e.currentTarget === e.target` is the only possibility).


I will say, I have generally used e.target as a JS dev for many years without problems. The typing on it throws me off here because `EventTarget` seems too generic. I would think we know it's a DOM Node at least? I got flustered when I saw the only properties were related to adding/removing event listeners. Historically, you'd use `e.target` more than `e.currentTarget` because you would have a single listener up high for performance reasons - like on the document element - and you'd use `e.target` to figure out which element was actually interacted with.

If you still want to use `e.target`, you can use `e.target instanceof HTMLInputElement` or whatever element you're looking for to refine the type on it, if you need to use that property.

**Reply by caseyjramirez** [reply]
You're a life savor. The force is strong with you.
